iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalGrammar.java

```diff
@@ -151,6 +163,9 @@ private static void lexical(LexerlessGrammarBuilder b) {
 
     b.rule(EOF).is(b.token(GenericTokenType.EOF, b.endOfInput())).skip();
 
+    b.rule(QUOTED_STRING_REGEX).is(b.regexp("'[^']*+'"));
+    b.rule(STRING_NO_SPACING_REGEX).is(b.regexp("(?:\\$|\\\\\"|[^\"$])++"));
```

Comment:
The regex looks scary, but it works. I'm not sure why so many backslashes are needed before a double quote 🤔